service/controller/resource/cloudconfigblob/create.go

```diff
@@ -0,0 +1,178 @@
+		} else if IsBootstrapCRNotReady(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "bootstrap CR is not ready")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling reconciliation")
+			reconciliationcanceledcontext.SetCanceled(ctx)
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var containerURL azblob.ContainerURL
+	{
+		containerURL, err = r.getContainerURL(ctx, credentialSecret, key.ClusterID(&azureMachinePool), key.StorageAccountName(&azureMachinePool))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring container object %#q contains bootstrap config", blobName))
+
+		_, err = blobclient.PutBlockBlob(ctx, blobName, payload, &containerURL)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured container object %#q contains bootstrap config", blobName))
+	}
+
+	return nil
+}
+
+// getCloudConfigFromBootstrapSecret returns the Bootstrap cloud config from the Bootstrap secret.
+func (r *Resource) getCloudConfigFromBootstrapSecret(ctx context.Context, machinePool *expcapiv1alpha3.MachinePool) (string, error) {
+	var err error
+	var bootstrapSecretName string
+	{
+		bootstrapSecretName, err = r.getBootstrapSecretName(ctx, machinePool)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+	}
+
+	var cloudconfigSecret corev1.Secret
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("Trying to find Secret containing bootstrap config %#q", bootstrapSecretName))
+		err := r.ctrlClient.Get(ctx, client.ObjectKey{Namespace: machinePool.Namespace, Name: bootstrapSecretName}, &cloudconfigSecret)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+	}
+
+	return string(cloudconfigSecret.Data[key.CloudConfigSecretKey]), nil
+}
+
+// getBootstrapSecretName will try to find Ignition CRs instead of Spark CRs when Ignition Operator is deployed.
+// It tries to find a Bootstrap CR which is named after the MachinePool. We may want to change it so we use `MachinePool.Spec.Template.Spec.Bootstrap`.
+func (r *Resource) getBootstrapSecretName(ctx context.Context, machinePool *expcapiv1alpha3.MachinePool) (string, error) {
+	var sparkCR corev1alpha1.Spark
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("Trying to find Bootstrap CR %#q", machinePool.Name))
+		err := r.ctrlClient.Get(ctx, client.ObjectKey{Namespace: machinePool.Namespace, Name: machinePool.Name}, &sparkCR)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+
+		if !sparkCR.Status.Ready || sparkCR.Status.DataSecretName == "" {
+			return "", microerror.Mask(bootstrapCRNotReady)
+		}
+	}
+
+	return sparkCR.Status.DataSecretName, nil
+}
+
+func (r *Resource) getContainerURL(ctx context.Context, credentialSecret *v1alpha1.CredentialSecret, resourceGroupName, storageAccountName string) (azblob.ContainerURL, error) {
```

Comment:
This was taken from `controllercontext` in other resources/handlers. I think it's fine retrieving it from the API. WDYT?

Reply:
Yeah, at least for now. We can iterate here later if we need to improve it somehow.